### PR TITLE
Changed regex for add format and added JUnit tests

### LINKED ISSUE #81: Update regex for add format to remove required order for arguments
Right now order for adding a task must be -d -t, and it cannot be -t -d

### PR DESCRIPTION
**Pull request by heekit73098**
Order does not matter now and command should throw exception for any invalid input

### CODE REVIEW

File: src/main/java/seedu/duke/parsers/AddParser.java [hunk 1]
```diff
@@ -38,16 +41,31 @@ public AddParser() {
...
         final String moduleDescription = parsedArguments.get(MODULE_DESCRIPTION);
+        final String taskDescriptionTwo = parsedArguments.get(TASK_DESCRIPTION_TWO);
+        String invalid = parsedArguments.get(INVALID);
+        String taskDescription = parsedArguments.get(TASK_DESCRIPTION);
+        if (invalid.isEmpty()) {
+            invalid = NULL_FIELD;
+        }
+        boolean isInvalid = ((!Objects.equals(invalid, NULL_FIELD))
+                || ((!Objects.equals(taskDescription, NULL_FIELD))
+                && (!Objects.equals(taskDescriptionTwo, NULL_FIELD))));
```
Comment (ngys117) [inline src/main/java/seedu/duke/parsers/AddParser.java]: Perhaps Objects.isNull(object) can replace Objects.equals if just checking whether the object is null. Also, not directed only at the usage here, but is there a need to assign null to a constant?

File: src/main/java/seedu/duke/parsers/AddParser.java [hunk 2]
```diff
@@ -15,19 +15,22 @@
...
     // TODO: Add support for -mod argument when integrating Task and Module classes with one another
-    private static final String ADD_FORMAT = "\\s*(\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))"
-            + "(\\s+(-d\\s+\\\"(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
-            + "(?<estimatedWorkingTime>([^\\\"]*))\\\"))?|\\/m\\s+"
-            + "(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|$))(\\s+(-d\\s+\\\"(?<moduleDescription>.+)\\\"))?)";
+    private static final String ADD_FORMAT = "\\s*((\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))(\\s+(-d\\s+\\\""
+            + "(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
+            + "(?<estimatedWorkingTime>([^\\\"]*))\\\")(?=(\\s+-d\\s+)|$))?(\\s+(-d\\s+\\\""
+            + "(?<taskDescription2>([^\\\"]*))\\\"))?|\\/m\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|\\s+.*$))"
+            + "(\\s+(-d\\s+\\\"(?<moduleDescription>.+)\\\"))?))(?<invalid>.*)";
```
Comment (ngys117) [inline src/main/java/seedu/duke/parsers/AddParser.java]: Regex works when adding tasks and ignores description and working time, but fails when trying to add modules using the following format `add /m MODULE_NAME`. Not sure if intentional.

**Review by ngys117 — CHANGES_REQUESTED** [review]
Seems mostly fine, check comments for suggested changes.

File: src/main/java/seedu/duke/parsers/AddParser.java (hunk 2 above)
Comment (heekit73098) [reply src/main/java/seedu/duke/parsers/AddParser.java]: yup i fixed the regex and added junit tests for add module so should be fine

File: src/main/java/seedu/duke/parsers/AddParser.java (hunk 1 above)
Comment (heekit73098) [reply src/main/java/seedu/duke/parsers/AddParser.java]: changed to isNull and NULL_FIELD to just null

**Review by ngys117 — APPROVED** [review]

**Comment by ngys117** [comment]
LGTM

**Pull request merged**